google/cloud/bigquery/table.py

```diff
@@ -999,6 +1000,24 @@ def table_constraints(self) -> Optional["TableConstraints"]:
             table_constraints = TableConstraints.from_api_repr(table_constraints)
         return table_constraints
 
+    @property
+    def external_catalog_table_options(self):
```

Comment:
Here we might want to return an `ExternalCatalogTableOptions` object, so the user get the same thing as their input, instead of a dictionary representation of it. [Here is a similar example](https://github.com/googleapis/python-bigquery/blob/255472359f3ed6b6cee06039ebe9059607fd9894/google/cloud/bigquery/job/query.py#L339-L352).

Reply:
This makes sense. I can see where something like this could be better represented by an object not a dict but where other items might be fine as just a dict.

Good call.

Will do.